src/escape/escape.c

```diff
@@ -204,6 +209,26 @@ int main(int argc, char *argv[]) {
                         break;
 
                 case ACTION_UNESCAPE:
+                        if (arg_template) {
+                                _cleanup_free_ char *instance, *template;
+
+                                r = unit_name_to_instance(*i, &instance);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to extract instance: %m");
+                                        goto finish;
+                                }
+                                r = unit_name_template(*i, &template);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to extract template: %m");
+                                        goto finish;
+                                }
+                                if (strcmp(arg_template, template)) {
+                                        log_error("Unit %s template %s does not match specified template %s.", *i, template, arg_template);
+                                        goto finish;
+                                }
+                                strcpy(*i, instance);
```

Comment:
No, things should be arranged so that all memory is freed at the end. But let's maybe leave those details until the end, after the semantics have been clarified.